extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -122,7 +129,9 @@ class AmpYoutube extends AMP.BaseElement {
       this.buildImagePlaceholder_();
     }
 
-    installVideoManagerForDoc(this.getAmpDoc());
+    const ampdoc = this.getAmpDoc();
+    installVideoManagerForDoc(ampdoc);
+    videoManagerForDoc(this.win.document).register(this);
```

Comment:
Any reason why `videoManagerForDoc` is using `this.win.document` instead of `ampdoc`? Using `win.document` might break YouTube inside shadow-doc/embedded use cases. You can test these manually by using the `/pwa/` and `/a4a/` prefixes before any example to see how it works. For example to test the `amp-youtube.amp.max.html` example this would be the URLs:

http://localhost:8000/pwa/examples/amp-youtube.amp.max.html
http://localhost:8000/a4a/examples/amp-youtube.amp.max.html

The only gotcha for the `pwa` example you'd probably need to make sure the service worker installed so it doesn't redirect you to the actual example instead of embedding it into shadow-doc

You could also possibly use `this.element` for any `_ForDoc` service call. 

cc/ @dvoytenko for any mis-info I am spreading 😄